packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -60,7 +64,12 @@ export type SelectedNetworkControllerGetSelectedNetworkStateAction = {
 
 export type SelectedNetworkControllerGetNetworkClientIdForDomainAction = {
   type: typeof SelectedNetworkControllerActionTypes.getNetworkClientIdForDomain;
-  handler: (domain: string) => NetworkClientId;
+  handler: (domain: string) => NetworkClientId | undefined;
+};
+
+export type SelectedNetworkControllerGetNetworkClientIdForMetamaskAction = {
+  type: typeof SelectedNetworkControllerActionTypes.getNetworkClientIdForMetamask;
+  handler: () => NetworkClientId;
```

Comment:
Nit: The handlers here could be coupled to the methods they're derived from, so that these manual follow-up updates are unnecessary.

```suggestion
  handler: SelectedNetworkController['getNetworkClientIdForDomain'];
};

export type SelectedNetworkControllerGetNetworkClientIdForMetamaskAction = {
  type: typeof SelectedNetworkControllerActionTypes.getNetworkClientIdForMetamask;
  handler: SelectedNetworkController['getNetworkClientIdForMetamask'];
```

Same for the next one as well:
```diff
 export type SelectedNetworkControllerSetNetworkClientIdForDomainAction = {
   type: typeof SelectedNetworkControllerActionTypes.setNetworkClientIdForDomain;
-  handler: (domain: string, NetworkClientId: NetworkClientId) => void;
+  handler: SelectedNetworkController['setNetworkClientIdForDomain'];
 };
```

Reply:
Done in latest!